### PR TITLE
[JENKINS-35570] List of computer's names extracted to Util

### PR DESCRIPTION
**Pull request by pjanouse**
Please review, txh. Pavel

### CODE REVIEW

File: core/src/main/java/hudson/Util.java [hunk 1]
```diff
@@ -1624,6 +1626,21 @@ public static Properties loadProperties(@Nonnull String properties) throws IOExc
         return p;
     }
 
+    /**
+     * @return The list of strings of computer names (excluding master)
+     * @since TODO
+     */
+    @Nonnull
+    public static List<String> getComputerNames() {
```
Comment (olivergondza) [inline core/src/main/java/hudson/Util.java]: I do not think the method belongs to `Util`, perhaps `Nodes` is a better option. Also, the clients seem to care more about nodes than computers.

File: test/src/test/java/hudson/UtilTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * @author pjanouse
+ */
+public class UtilTest {
+    @Rule
+    public final JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testCetComputerNames() throws Exception {
```
Comment (olivergondza) [inline test/src/test/java/hudson/UtilTest.java]: NIT: typo and redundant `test` prefix.

File: core/src/main/java/hudson/Util.java (hunk 1 above)
Comment (pjanouse) [inline core/src/main/java/hudson/Util.java]: `ComputerSet` would be even better.

**Comment by oleg-nenashev** [comment]
👍

File: core/src/main/java/hudson/model/ComputerSet.java [hunk 3]
```diff
@@ -410,6 +411,21 @@ public void doRun() {
...
     }
 
+    /**
+     * @return The list of strings of computer names (excluding master)
+     * @since TODO
+     */
+    @Nonnull
+    public static List<String> getComputerNames() {
+        final ArrayList<String> names = new ArrayList<String>();
+        for (Computer c : Jenkins.getActiveInstance().getComputers()) {
```
Comment (olivergondza) [inline core/src/main/java/hudson/model/ComputerSet.java]: `getActiveInstance` is deprecated, please use good old `getInstance`.

Comment (daniel-beck) [inline core/src/main/java/hudson/model/ComputerSet.java]: @oleg-nenashev Didn't we revert the change to `getInstance` for compatibility reasons that was supposed to make `getActiveInstance` obsolete?

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ComputerSet.java]: Yes, we did. But we do not really expect null instance to happen on master for extension points

**Comment by oleg-nenashev** [comment]
re-:+1:

**Comment by oleg-nenashev** [comment]
@olivergondza WDYT?

File: test/src/test/java/hudson/model/ComputerSetTest.java [hunk 4]
```diff
@@ -76,4 +82,15 @@ public void nodeOfflineCli() throws Exception {
...
+
+    @Test
+    public void getComputerNames() throws Exception {
+        assertThat(ComputerSet.getComputerNames(), is(empty()));
+        j.createSlave("aNode", "", null);
+        assertThat(ComputerSet.getComputerNames(), hasSize(1));
+        assertThat(ComputerSet.getComputerNames(), contains("aNode"));
+        j.createSlave("anAnotherNode", "", null);
+        assertThat(ComputerSet.getComputerNames(), hasSize(2));
+        assertThat(ComputerSet.getComputerNames(), containsInAnyOrder("aNode", "anAnotherNode"));
```
Comment (olivergondza) [inline test/src/test/java/hudson/model/ComputerSetTest.java]: `containsInAnyOrder` actually check if the collections are of the same size and contain same elements regardless of their position in the collection. IOW, `hasSize` should not be needed as long as we compare using `containsInAnyOrder`, which I recommend in both cases here.

**Comment by olivergondza** [comment]
Added one more minor comment. Should be merged once addressed.

**Comment by oleg-nenashev** [comment]
@pjanouse gentle ping (1 minor comment)

**Comment by pjanouse** [comment]
Fixed, sry for the delay... Pavel

**Comment by oleg-nenashev** [comment]
Test failure is unrelated
